### DISCUSSION TITLE
Simulating MouseUp MouseDown

### DISCUSSION THREAD

**Discussion by damiensellier** [RomanKubiak/ctrlr#280] [Q&A]
Hi Everybody,
Hope you a nice sunny day like I did. 

Do you know if there's a way to simulate a MouseUp, MouseDown, DoubleClick... via [JUCE](https://docs.juce.com/master/classMouseEvent.html) from a LUA script?

I still miss something as well, I want to disable the value change capability when you click on the value displayed at the bottom of the slider, and ot opens a blank field for the value. i don't want that because value can be anything unregistered. If anyone knows that would be awesome.

Thanks in advance, have a nice week.

Damien

PS: If someone is wondering how to disable the value change on mouseScroll while hovering the mods, the only way I found is to change "uiSliderMouseWheelInterval" to 0 and it's perfect.

**Comment by DRBAUS** [comment]
I would set it to "NoTextBox" and create a label that displays the value....but that's probably the beginner's way of doing it and there might be a formula for it.

**Reply by damiensellier** [reply]
That's a nice idea for a few sliders but i have 60 on my project so I need 60 extra labels and formulas to fit the current value in it. I'll search the JUCE doc to find if there's an hidden command to disable that feature and let you all know.

**Comment by dnaldoog** [comment]
Hi Damien!

One way to suppress user input would be to create a uiImage filled with a transparent PNG image, say 16 x 16 pixels and then place that uiImage in a layer above the control that you don’t  want to be clicked on. This is also useful when you have a uiButtonImage say for example an on off light that changes its image from a lua script, but you don’t  want to user to Inadvertently click on it.

**Reply by damiensellier** [reply]
Oh yeah i see we can use the transparent image as a mask. that's a nice trick!
